syspage.c

```diff
@@ -39,27 +39,30 @@ void syspage_init(void)
 
 	syspage_common.syspage->maps = NULL;
 	syspage_common.syspage->progs = NULL;
-	syspage_common.syspage->size = sizeof(syspage_t);
 	syspage_common.syspage->console = console_default;
 
-	syspage_common.heapTop = (char *)syspage_common.syspage + sizeof(syspage_t);
+	syspage_common.heapTop = (void *)ALIGN_ADDR((addr_t)syspage_common.syspage + sizeof(syspage_t), 4);
```

Comment:
This is platform independent code. Maybe you should align to the biggest standard type, instead of 4?
```suggestion
	syspage_common.heapTop = (void *)ALIGN_ADDR((addr_t)syspage_common.syspage + sizeof(syspage_t), sizeof(long long);
```

Reply:
I agree with you. Changed to `sizeof(long long)`.